applications/nrf_desktop/doc/hid_provider_system_ctrl.rst

```diff
@@ -39,85 +62,64 @@ The module selects the :ref:`CONFIG_DESKTOP_HID_KEYMAP <config_desktop_app_optio
-The :ref:`nrf_desktop_hid_state` receives the event and fills it with its own implementation of callbacks from the HID state API (:c:struct:`hid_state_api`).
-After that process, the modules can communicate by callbacks.
-The module also subscribes to the :c:struct:`button_event` event to get information about the button presses.
-The module sends :c:struct:`hid_report_event` events to an appropriate subscriber when it is requested by the :ref:`nrf_desktop_hid_state`.
-
-.. note::
-    The HID report formatting function must work according to the HID report descriptor (``hid_report_desc``).
-    The source file containing the descriptor is provided by the :ref:`CONFIG_DESKTOP_HID_REPORT_DESC <config_desktop_app_options>` Kconfig option.
-
-Linking input data with the right HID report
-============================================
+The module is used by :ref:`nrf_desktop_hid_state` as a HID input report provider for the HID system control input report.
+On initialization, the module submits the :c:struct:`hid_report_provider_event` event to establish two-way callbacks between the |hid_state| and the HID report provider.
 
-Out of all available input data types, the module collects button events.
-The button events are stored in the :c:struct:`report_data` structure.
-The ``button_event`` is the source of this type of data.
-
-To indicate a change to this input data, the module overwrites the value that is already stored.
-
-Since keys on the board can be associated with a HID usage ID and thus be part of different HID reports, the first step is to identify if the key belongs to a HID report that is provided by this module.
-This is done by obtaining the key mapping from the :ref:`nrf_desktop_hid_keymap`.
-
-Once the mapping is obtained, the application checks if the report to which the usage belongs is connected:
-
-* If the report is connected and the :ref:`nrf_desktop_hid_eventq` instance is empty, the module stores the report and calls the ``trigger_report_send`` callback from the :c:struct:`hid_state_api` to notify the :ref:`nrf_desktop_hid_state` about the new data.
-* If the report is not connected or the :ref:`nrf_desktop_hid_eventq` instance is not empty, the value is enqueued in the :ref:`nrf_desktop_hid_eventq` instance.
-
-The difference between these operations is that storing a value onto the queue (second case) preserves the order of input events.
-See the following section for more information about storing data before the connection.
-
-Storing input data before the connection
-========================================
-
-The button data is stored before the connection.
-
-The reason for this operation is to allow to track key presses that happen right after the device is woken up, but before it can connect to the HID host.
+Handling keypresses
```

Comment:
We could use .. include:: for this section as the same is used in the hid_provider_consumer_ctrl.rst file - Handling keypresses.